docs/Developer Documentation/Logger/index.md

```diff
@@ -40,7 +42,16 @@ The `Logger` interface contains only methods required by any SDL component to pe
 The message loop thread is needed to avoid significant performance degradation at run time as logging calls are blocking calls and might take a significant amount of time. `LoggerImpl::PushLog` is a non-blocking call. It will put the log message into the queue and returns immediately.
 
 
-If `ThirdPartyLoggerInterface` supports non blocking threaded logging, minor changes in `LoggerImpl` can be made with `use_message_loop_thread = false`. 
+If `ThirdPartyLoggerInterface` supports non blocking threaded logging, minor changes in `LoggerImpl` can be made with `use_message_loop_thread = false`.
+
+### Configurable time before shutdown
+
+To prevent missing logs after SDL shutdown, after receiving IGNITION_OFF signal SDL dumps all logs into the file system before shutdown. To control logs flushing process SDL logger has additional ini file options:
+
+ * Write all logs to file system before shutdown(using `LoggerImpl::Flush` function)
```

Comment:
"write all logs" is misleading info, it does not guarantee all logs are written to the disk with a small maximum wait time.

Reply:
the string FlushLogMessagesBeforeShutdown in the sdl proposal is better

Reply:
I didn't quite get what you mean. Per the proposal FlushLogMessagesBeforeShutdown is Boolean, not string. Could you please clarify this?
I can suggest to change "**Write all logs to file system before shutdown**" to "**Write logs from queue to file system before shutdown**"

Reply:
I like the following from the proposal
option to flush log messages before shutdown.
option that specifies maximum time before shutdown.

Reply:
@VladSemenyuk the keyword "all" that causes the confusion, my understanding is everything in the logs will be written to disk, that is all. but it might not be the case

Reply:
@yang1070 Done